containerd-shim-v2/service.go

```diff
@@ -548,7 +548,27 @@ func (s *service) Connect(ctx context.Context, r *taskAPI.ConnectRequest) (*task
 }
 
 func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (*ptypes.Empty, error) {
-	return nil, errdefs.ErrNotImplemented
+	s.Lock()
+	defer s.Unlock()
+
+	if len(s.containers) != 0 {
+		return empty, nil
+	}
+
+	defer os.Exit(0)
+
+	err := s.sandbox.Stop()
```

Comment:
What if the sandbox has been already stopped and deleted? We should check for that, and also check that `s.sandbox` is valid before we use it.

Reply:
No, this case will not happen unless there is a bug. Since the Shutdown() api is the only one entrance of stop and delete the sandbox, once it stop and delete it, this shimv2 process will also
exit and there is no chance for a shimv2 process running with sandbox has been stopped.